Sprint 1/step 2

yakovlevyaroslav opened the pull request.

yakovlevyaroslav committed:
первый этап первого спринта

yakovlevyaroslav committed:
убрал из .gitignore /public и запушил ее

yakovlevyaroslav committed:
поменял теги компоненов в App.jsx на самозакрывающиеся, так как в них нет содержимого

yakovlevyaroslav committed:
поменял инлайновые стили на классы css

yakovlevyaroslav committed:
Внес правки по ревью:
- оставил в стилять только вертикальный скролл
- сделал перестройку интерфейся от 980px, а не от 1200px, чтобы он дольше оставался горизонтальным
- добавил булкам в компоненте BurgerConsctructor type, надписи верх и низ, и сделал так чтобы булка была одинаковая для верха и низа

yakovlevyaroslav committed:
Сделал все задания из /step-2

yakovlevyaroslav committed:
Добавил проверку типов пропсов с помощью PropTypes

gennady-bars commented on src/components/app/App.jsx:
```diff
@@ -0,0 +1,89 @@
+    const fetchBurgerIngredients = async () => {
+      try {
+        const response = await fetch(URL_INGREDIENTS);
+        if (!response.ok) {
+          throw new Error(`Ошибка! статус: ${response.status}`);
+        }
+        const data = await response.json();
```
Отлично, что проверяете ответ на `ok`

gennady-bars commented on src/components/app/App.jsx:
```diff
@@ -0,0 +1,89 @@
+        }
+        const data = await response.json();
+        setIngredients(data.data);
+      } catch (error) {
```
Отлично, что ловите возможные ошибки в конце каждого запроса к серверу

gennady-bars commented on src/components/app/App.jsx:
```diff
@@ -0,0 +1,89 @@
+    };
+
+    fetchBurgerIngredients();
+  }, []);
```
Отлично, что используете пустой массив зависимостей `[]`, чтобы `useEffect` сработал 1 раз

gennady-bars commented on src/components/app/App.jsx:
```diff
@@ -0,0 +1,89 @@
+      setSelectedIngredient([]);
+    }
+    setIsModalOpen(true);
+    document.body.style.overflow = 'hidden';
```
`    document.body.style.overflow = 'hidden';
`
**Можно лучше**

Это не требуется в `App`. Легче всего это сделать внутри `Modal`. Там будет обработка `Escape`, где все это реализовать можно

gennady-bars commented on src/components/app/App.jsx:
```diff
@@ -0,0 +1,89 @@
+      {isModalOpen && 
+        <>
+          <Modal type={typeModal} onClose={closeModal} propIngredient={selectedIngredient} />
+          <ModalOverlay onClose={closeModal}/>
```
`ModalOverlay` нужно поместить внутрь `Modal`, чтобы в будущем не дублировать его где-то еще

gennady-bars commented on src/components/modal/Modal.jsx:
```diff
@@ -0,0 +1,50 @@
+      <div className={`${modalStyles.modal} p-10`}>
+        <div className={`${modalStyles['modal-header']}`}>
+          {
+            type === 'ingredient' ?
```
` type === 'ingredient'`

Это харкод под какой-то ингредиент

Внутри компонента `Modal` не должно быть **хардкода** строк и компонентов. `IngredientDetails` и `OrderDetails` нужно передавать в качестве `children`.

gennady-bars commented on src/components/modal/Modal.jsx:
```diff
@@ -0,0 +1,50 @@
+          {
+            type === 'ingredient' ?
+              <div className={`pt-3 text text_type_main-large`}>
+                Детали заказа
```
Внутри компонента `Modal` не должно быть **хардкода** строк и компонентов. `IngredientDetails` и `OrderDetails` нужно передавать в качестве `children`.

Нужно передавать `title` в пропсы

gennady-bars commented on src/components/modal/Modal.jsx:
```diff
@@ -0,0 +1,50 @@
+          </>
+          :
+            <OrderDetails />
+        }
```
Внутри компонента `Modal` не должно быть **хардкода** строк и компонентов. `IngredientDetails` и `OrderDetails` нужно передавать в качестве `children`.

gennady-bars commented on src/components/modal-overlay/ModalOverlay.jsx:
```diff
@@ -0,0 +1,35 @@
+const modalOverlay = document.getElementById('modal-overlay');
+
+function ModalOverlay({onClose}) {
+  const handleEscKey = (event) => {
```
Обработчик `Escape` нужно поместить внутрь `useEffect` (внутрь `Modal`), чтобы не менялась ссылка на функцию при перерисовке компонента.

gennady-bars commented on src/components/modal-overlay/ModalOverlay.jsx:
```diff
@@ -0,0 +1,35 @@
+    }
+  };
+
+  useEffect(() => {
```
Обработчик `Escape` нужно переместить в `Modal`. Это его логика, а не оверлея. Оверлей - это просто темное пространство вокруг попапа

gennady-bars commented on src/components/modal-overlay/ModalOverlay.jsx:
```diff
@@ -0,0 +1,35 @@
+    };
+  }, []);
+
+  return createPortal ( 
```
`createPortal` будет только в `Modal`. Вся верстка целиком будет уходить в один портал

gennady-bars commented on public/index.html:
```diff
@@ -0,0 +1,45 @@
+    <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="root"></div>
+    <div id="modal"></div>
+    <div id="modal-overlay"></div>
```
`    <div id="modal-overlay"></div>
`

Это не нужно
* `createPortal` будет только в `Modal`. Вся верстка целиком будет уходить в один портал

gennady-bars commented on src/components/order-details/OrderDetails.jsx:
```diff
@@ -0,0 +1,23 @@
+import orderDetailsStyles from  './orderDetails.module.css';
+
+function orderDetails() {
```
`orderDetails`

Кастомные `компоненты` принято называть в `PascalCase` (с **большой буквы**)

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:



* Отлично, что ловите возможные ошибки в конце каждого запроса к серверу
* Отлично, что используете пустой массив зависимостей `[]`, чтобы `useEffect` сработал 1 раз

но есть некоторые недочеты:



* Видна граница оверлея при скролле вниз https://disk.yandex.ru/i/pNFyvjpn1qH4rQ  Модальное окно должно быть всегда ровно посередине экрана. Для этого нужно указать ему и оверлею в `css` позицию `fixed`.
* `ModalOverlay` нужно поместить внутрь `Modal`, чтобы в будущем не дублировать его где-то еще
* Внутри компонента `Modal` не должно быть **хардкода** строк и компонентов. `IngredientDetails` и `OrderDetails` нужно передавать в качестве `children`.
* Обработчик `Escape` нужно переместить в `Modal`. Это его логика, а не оверлея. Оверлей - это просто темное пространство вокруг попапа
* `createPortal` будет только в `Modal`. Вся верстка целиком будет уходить в один портал
* Кастомные `компоненты` принято называть в `PascalCase` (с **большой буквы**)

yakovlevyaroslav committed:
Поправил замечания из pull request:
- перенес ModalOverlay в Modal и оставил только один компонент с использованием createPortal
- убрал весь хардкод и лишние пропсы в компоненте Modal
- перенес логику обработки нажатия кнопки Escape в Modal с использованием useCallback, чтобы handleEscKey не создавалась заного при каждом рендере компонента
- компоненты IngredientDetails и OrderDetails подключаются в App и отправляются в Modal через проп children
- проверил, чтобы все компоненты были в PascalCase

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.

yakovlevyaroslav commented:
> Поздравляю! Ваша работа принята.
> 
> Вы отлично потрудились.
> 
> Удачного дальнейшего обучения.

Спасибо за подробный разбор!
